framework/util/driver_info.cpp

```diff
@@ -172,6 +281,48 @@ bool GetDriverInfo(std::string& driver_info, format::ApiFamilyId api_family)
     return driver_info_available;
 }
 
+bool RegistryDxDriverVersion(std::string& driver_info, const std::vector<LUID>& adapter_luids)
+{
+    bool driver_info_found = false;
+
+    HKEY    dx_key_handle      = nullptr;
+    DWORD   num_of_adapters    = 0;
+    DWORD   sub_key_max_length = 0;
+    LSTATUS return_code        = GetRegSubkeys(dx_key_handle, num_of_adapters, sub_key_max_length);
+    if (return_code == ERROR_SUCCESS)
+    {
+        return_code = GetRegData(dx_key_handle, num_of_adapters, sub_key_max_length, driver_info, adapter_luids);
+        if (return_code == ERROR_SUCCESS)
+        {
+            driver_info_found = true;
+        }
+    }
+
+    return driver_info_found;
+}
+
+#endif
+
+std::string ConvertDWORDtoVersionNumber(uint64_t dword)
+{
+    uint32_t    version[4]  = {};
+    std::string str_version = "";
+    version[0]              = (unsigned int)((dword & 0xFFFF000000000000) >> 16 * 3);
+    version[1]              = (unsigned int)((dword & 0x0000FFFF00000000) >> 16 * 2);
+    version[2]              = (unsigned int)((dword & 0x00000000FFFF0000) >> 16 * 1);
+    version[3]              = (unsigned int)((dword & 0x000000000000FFFF));
+    str_version = std::to_string(version[0]) + "." + std::to_string(version[1]) + "." + std::to_string(version[2]) +
+                  "." + std::to_string(version[3]) + ".";
```

Comment:
Final period in version number could be removed: `Driver Version: 31.0.12029.4013.` -> `Driver Version: 31.0.12029.4013`